### DISCUSSION TITLE
PnP Search v4 extensibility webpart

### DISCUSSION THREAD

**Discussion by Shubhajit00450** [microsoft-search/pnp-modern-search#3155] [Q&A]
Hi,

I am trying to extend the pnp search v4.8 webpart to display documents on a search Page where we are using <pnp-filepreview > reusable web components to customize the template. The template shows the preview of the documents onClick of the document thumbnail. However, we want it to open on hovering the preview. Also, on Hover the document (ppt, word, excel) should open in sliding way instead of opening the complete document. Is there any control or action available to achieve this behaviour?

Kindly, suggest

**Comment by wobba** [comment]
No control in the existing controls. You would have to build this extensibility yourself.
